src/runtime_src/core/common/CMakeLists.txt

```diff
@@ -45,6 +45,7 @@ target_link_libraries(xrt_coreutil
   )
 
 if (NOT WIN32)
+set_target_properties(xrt_coreutil PROPERTIES INTERFACE_LINK_LIBRARIES dl)
```

Comment:
I don't think clients need link with -dl.  I don't see any usage of dlsym, dlopen, etc in coreutils, there is simply header that expose / wrap these functions but that it for internal XRT use only.  External clients of coreutils are not required to link with -dl.

I see travis link errors on arm64, are they related to this target requirement?

Reply:
You are seems to be right, I am trying to reproduce the Travis issue.

Reply:
Currently, I've found that

`/usr/bin/g++-8 -fPIC -g  -shared -Wl,-soname,liboclxdp.so.2 -o liboclxdp.so.2.6.0 CMakeFiles/oclxdp.dir/profile/plugin/ocl/ocl_power_profile.cpp.o CMakeFiles/oclxdp.dir/profile/plugin/ocl/ocl_profiler.cpp.o CMakeFiles/oclxdp.dir/profile/plugin/ocl/xocl_plugin.cpp.o CMakeFiles/oclxdp.dir/profile/plugin/ocl/xocl_profile.cpp.o CMakeFiles/oclxdp.dir/profile/plugin/ocl/xocl_profile_cb.cpp.o -Wl,-rpath,/XRT/build/Debug/runtime_src/xocl:/XRT/build/Debug/runtime_src/xdp:/XRT/build/Debug/runtime_src/xrt:/XRT/build/Debug/runtime_src/core/common: ../xocl/libxilinxopencl.so.2.6.0 libxdp.so.2.6.0 ../xrt/libxrt++.so.2.6.0 -ldl -lcrypt -lrt ../core/common/libxrt_coreutil.so.2.6.0 -Wl,-Bstatic -lboost_filesystem -lboost_system -Wl,-Bdynamic -lpthread -luuid`


works, whereas

`/usr/bin/g++-8 -fPIC -g  -shared -Wl,-soname,liboclxdp.so.2 -o liboclxdp.so.2.6.0 CMakeFiles/oclxdp.dir/profile/plugin/ocl/ocl_power_profile.cpp.o CMakeFiles/oclxdp.dir/profile/plugin/ocl/ocl_profiler.cpp.o CMakeFiles/oclxdp.dir/profile/plugin/ocl/xocl_plugin.cpp.o CMakeFiles/oclxdp.dir/profile/plugin/ocl/xocl_profile.cpp.o CMakeFiles/oclxdp.dir/profile/plugin/ocl/xocl_profile_cb.cpp.o -Wl,-rpath,/XRT/build/Debug/runtime_src/xocl:/XRT/build/Debug/runtime_src/xdp:/XRT/build/Debug/runtime_src/xrt:/XRT/build/Debug/runtime_src/core/common: ../xocl/libxilinxopencl.so.2.6.0 libxdp.so.2.6.0 ../xrt/libxrt++.so.2.6.0 -Wl,-Bstatic -lboost_filesystem -lboost_system -Wl,-Bdynamic -lcrypt -lrt ../core/common/libxrt_coreutil.so.2.6.0 -ldl -lpthread -luuid`

produces linker errors in CI.

Reply:
Now it is clear to me, that `../core/common/libxrt_coreutil.so.2.6.0` already statically linked with `boost_filesystem` and has all its symbols inside, so their relative order matters here. I am not sure that this is how it is suppose to be.
I think that my initial change is not the case for this issue but only an reorder trigger.

Reply:
yeah, this is a bit odd, I don't really know either, it will take some digging.  But since -dl is not a requirement when using coreutils, can you just remove that change?

Reply:
sure